tfprotov5/client_capabilities.go

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tfprotov5
+
+// ClientCapabilities allows Terraform to publish information regarding optionally supported
+// protocol features, such as forward-compatible Terraform behavior changes.
+type ClientCapabilities struct {
+	// DeferralAllowed signals that the Terraform client is able to
+	// handle deferred responses from the provider.
+	DeferralAllowed bool
```

Comment:
Nit: For future travelers, I wonder if it would be worth calling out which RPCs this applies to, otherwise I worry about this struct having a lot of fields that only sometimes apply to whatever RPC its attached to.

Another option would be creating RPC-specific `ClientCapabilities` struct types (e.g. `ConfigureProviderClientCapabilities` 😅 ) to protect the exported API from changing in a future where the protocol potentially introduces RPC-specific capabilities. We had opted for the protocol to share the message for now and changing that in the protocol is safe to split up as it only affects the generated code, but exported Go package code not so much.

Reply:
Yeah this is a good point, it feels very likely this would happen, especially with the design that the capabilities are always related to specific requests and not the client as a whole.

I'll duplicate all of these to structs as `{RPCName}ClientCapabilities` to protect us from that but keep the actual property name as `ClientCapabilities`.

cc @SBGoods -> just a heads up that there will be an upcoming change once I adjust this

Reply:
Split all of the structs up -> 3a4fbc8 ⚡